tests/test_models/test_inference/test_forecast.py

```diff
@@ -800,3 +836,109 @@ def test_forecast_subset_segments(self, model, transforms, example_tsds):
     )
     def test_forecast_subset_segments_failed_assertion_error(self, model, transforms, example_tsds):
         self._test_forecast_subset_segments(example_tsds, model, transforms, segments=["segment_2"])
+
+
+class TestForecastNewSegments:
+    """Test forecast on new segments.
+
+    Expected that target values are filled after prediction.
+    """
+
+    def _test_forecast_new_segments(self, ts, model, transforms, train_segments, prediction_size=5):
+        # create tsdataset with new segments
+        train_segments = list(set(train_segments))
+        forecast_segments = list(set(ts.segments) - set(train_segments))
+        train_ts = select_segments_subset(ts=deepcopy(ts), segments=train_segments)
+        test_ts = select_segments_subset(ts=deepcopy(ts), segments=forecast_segments)
+
+        # fitting
+        train_ts.fit_transform(transforms)
+        test_ts.transform(train_ts.transforms)
+        model.fit(train_ts)
+
+        # forecasting
+        import torch  # TODO: remove after fix at issue-802
+
+        torch.manual_seed(11)
+
+        forecast_ts = test_ts.make_future(future_steps=prediction_size, tail_steps=model.context_size)
+        forecast_ts = make_forecast(model=model, ts=forecast_ts, prediction_size=prediction_size)
+
+        # checking
+        forecast_df = forecast_ts.to_pandas(flatten=True)
+        assert not np.any(forecast_df["target"].isna())
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (CatBoostModelMultiSegment(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (LinearMultiSegmentModel(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (ElasticMultiSegmentModel(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (RNNModel(input_size=1, encoder_length=7, decoder_length=7, trainer_params=dict(max_epochs=1)), []),
+            (
+                MLPModel(input_size=2, hidden_size=[10], decoder_length=7, trainer_params=dict(max_epochs=1)),
+                [LagTransform(in_column="target", lags=[5, 6])],
+            ),
+        ],
+    )
+    def test_forecast_new_segments(self, model, transforms, example_tsds):
```

Comment:
May we check somehow that all the multisegment models are here, I mean if we add new model, this test should inform us somehow that we should add this model to it(the same for per-segment models)